Sming/gdb/gdbstub-cfg.h

```diff
@@ -178,7 +178,7 @@
  * Set to 0 to wait indefinitely.
  */
 #ifndef GDBSTUB_UART_READ_TIMEOUT
-#define GDBSTUB_UART_READ_TIMEOUT 2000
+#define GDBSTUB_UART_READ_TIMEOUT 0
 #endif
 
```

Comment:
To keep the instructions simpler, how about we do this:

To build for normal (command-line GDB) use: `make flash`
To build for eclipse: `make flash ENABLE_CONSOLE=0`

?

Reply:
Let's use `ENABLE_SYS_CONSOLE` instead of `ENABLE_CONSOLE`. In addition I would prefer the syscall stuff to be disabled by default. For the LiveSample we can enable it by default by writing in the Makefile-user.mk.
When  `ENABLE_SYS_CONSOLE` is enabled a warning message should be shown to inform the developers that this might interfere with visual debuggers, like Eclipse. And that it can be disabled using `make ENABLE_SYS_CONSOLE=0`.
What do you say?

Reply:
Or even better `ENABLE_GDB_CONSOLE` instead of `ENABLE_CONSOLE`. The name prompts that this is related to the special GDB console.

Reply:
> Allow also that setting to be changed during compilation so that adventurous developers can try it.

The simplest way is to just edit the `gdbstub-cfg.h` file directly to make changes as required (same goes for `gdbcmds`). I also have a note in the `readme.md` about using `USER_CFLAGS` to set these values for a particular project - I've done this for the `LiveDebug` sample as per your suggestions.

Are you propsing we add the abillty to change these directly from the `make` command line?

Reply:
> Are you proposing we add the ability to change these directly from the make command line?

Ok, I see. Then can you do the  the following?

1. GDBSTUB_ENABLE_SYSCALL=0 in Sming/gdb/gdbstub-cfg.h
2. In samples/LiveDebug/Makefile-user.mk add
```
# Comment the line below if you want to disable the GDB system calls
USER_CFLAGS += -DGDBSTUB_ENABLE_SYSCALL=1
```
3.  Update the README in LiveDebug to mention that change.